shared/types/background.spec.ts

```diff
@@ -0,0 +1,34 @@
+import type {
+  IsEquivalent,
+  ControllerStatePropertiesEnumerated,
+  FlattenedBackgroundStateProxy,
+} from './background';
+
+abstract class _ {
+  abstract readonly _: symbol;
+}
```

Comment:
This emits a runtime class. _Maybe_ the minifier/optimizer drops it, but I wouldn't always count on it. Can we use another mechanism?

Reply:
Would something like the following be acceptable?

```
diff --git a/development/webpack/webpack.config.ts b/development/webpack/webpack.config.ts
index e5034dfd99..5e2b7d51e4 100644
--- a/development/webpack/webpack.config.ts
+++ b/development/webpack/webpack.config.ts
@@ -268,7 +268,9 @@ const config = {
       },
       // own typescript, and own typescript with jsx
       {
-        test: /\.(?:ts|mts|tsx)$/u,
+        test: args.test
+          ? /\.(?:ts|mts|tsx)$/u
+          : /(?<!\.test|spec)\.(?:ts|mts|tsx)$/u,
         exclude: NODE_MODULES_RE,
         use: [tsxLoader, codeFenceLoader],
       },
```

Since tsc doesn't emit files into builds, as long as we're preventing test files from being processed by swc and webpack we shouldn't get any runtime artifacts. This exclusion is something we might want to add in any case just to be on the safe side, although it's clearly redundant as of now.

The ability to write type-level tests might come in handy as we progress with conversions, and the `_` abstract class is useful for this, while not being easily replaceable.

Reply:
Ah, I didn't even realize it was a test. I don't think we need the exclusion (and I'm pretty sure we have runtime code that pulls in stuff from test files anyway).

I just know that there are alternative ways to do what you are doing above `abstract class` in types only.

Reply:
This appears to work for our purposes here as well:

```suggestion
type _ = {
  readonly _: unique symbol;
}
```

I'm not sure if it will make `Expect` generalizable to all type inputs, but I'm happy to use it to replace `abstract class` here so we can avoid runtime artifacts -- just as a matter of principle. 

If there are other good ways for achieving this effect please let me know. You can't drop a hint like that and leave me hanging.😂